src/validations/html/rules/index.js

```diff
@@ -0,0 +1,50 @@
+import Validator from '../../Validator';
+import runRules from '../runRules';
+import Code from './Code';
+import MismatchedTag from './MismatchedTag';
+
+const errorMap = {
+  [Code.MISPLACED_CLOSE_TAG]: ({openTag, closeTag, match}) => ({
+    reason: 'misplaced-close-tag',
+    payload: {
+      open: openTag.name,
+      close: closeTag.name,
+      misplacement: match.row + 1,
+    },
+  }),
+  [Code.UNCLOSED_TAG]: ({openTag, closeTag}) => ({
+    reason: 'unclosed-tag-by-mismatched-close-tag',
```

Comment:
> Is there a reason not to just report this as a normal `unclosed-tag`?

Not really other than this was based on `mismatched-close-tag` but:

> Is it really useful to include information about where we found a mismatched closing tag?

Hmm, not really, since the error location already shows where the mismatch was found.

> Would it make more sense to still locate the error at the location of the opening tag?

Actually, the `RuleValidator` (and previous versions) always used the location of the closing tag as the error location. But I agree that the closing tag location + the existing `unclosed-tag` message should suffice.

Too much information in an error can easily overload students!